bot/skills/buktopuha.py

```diff
@@ -305,3 +408,157 @@ def start_buktopuha(update: Update, context: CallbackContext):
     )
+
+    existing_user = _db.find(user_id=update.effective_user.id)
+    if existing_user is None:
+        _db.add(user=update.effective_user, score=0)
+    else:
+        _db.game(user_id=update.effective_user.id)
+
+
+def show_nerds(update: Update, context: CallbackContext):
+    """Show leader board, I believe it should looks like smth like:
+
+                    3HaToKu BuKToPuHbI
+    ==================================================
+        score   |  games  |   wins  |     znatok
+    ------------+---------+---------+-----------------
+    100500      | 666     | 666     | egregors
+    9000        | 420     | 999     | getjump
+    --------------------------------------------------
+
+    """
+
+    logger.error(update)
+    # CSS is awesome!
+    # todo:
+    #  need to find out how to show board for mobile telegram as well
+    board = (
+        f"{'3HaToKu BuKToPuHbI'.center(52)}\n"
+        f"{'='*55}\n"
+        f"{'score'.center(12)} "
+        f"| {'games'.center(9)} "
+        f"| {'wins'.center(9)} "
+        f"| {'znatok'.center(16)} "
+        f"\n"
+        f"{'-'*12} + {'-'*9} + {'-'*9} + {'-'*16}\n"
+    )
+
+    znatoki = _db.find_all()
+    znatoki_length = len(znatoki)
+
+    for znatok in znatoki:
+        username = _get_username(znatok)
+        board += (
+            f"{str(znatok['total_score']).ljust(12)} "
+            f"| {str(znatok['game_counter']).ljust(9)} "
+            f"| {str(znatok['win_counter']).ljust(9)} "
+            f"| {username.ljust(16)}\n"
+        )
+
+    board += f"{'-'*55}"
+    try:
+        board_image, board_image_path = from_text_to_image(board, znatoki_length)
+    except (ValueError, RuntimeError, OSError) as ex:
+        logger.error("Cannot get image from text, znatoki error: %s", ex)
+        return
+
+    result: Optional[Message] = None
+
+    if znatoki_length <= ZNATOKI_LIMIT_FOR_IMAGE:
+        result = context.bot.send_photo(
+            chat_id=update.effective_chat.id,
+            photo=board_image,
+            disable_notification=True,
+        )
+    else:
+        result = context.bot.send_document(
+            chat_id=update.effective_chat.id,
+            document=board_image,
+            disable_notification=True,
+        )
+
+    cleanup_queue_update(
+        context.job_queue,
+        update.message,
+        result,
+        600,
+        remove_cmd=True,
+        remove_reply=False,
+    )
+
+    os.remove(board_image_path)
+
+
+def _get_username(h: Dict) -> str:
+    """Get username or fullname or unknown"""
+    m = h["meta"]
+    username = m.get("username")
+    fname = m.get("first_name")
+    lname = m.get("last_name")
+    return (
+        username
+        or " ".join(filter(lambda x: x is not None, [fname, lname]))
+        or "unknown"
+    )
+
+
+JPEG = "JPEG"
+EXTENSION = ".jpg"
+COLOR = "white"
+MODE = "L"
+FONT_SIZE = 12
+ZNATOKI_LIMIT_FOR_IMAGE = 25
+FONT = "firacode.ttf"
+
+
+def _create_empty_image(image_path, limit):
```

Comment:
All this stuff looks like a copy-paste. Is it from `top`? 
Maybe we should extract it in some sort of common utility? 
I mean text-to-image func, zum Beispiel.

Reply:
It is from roll. Not common enough for me to justify abstracting away. I usually separate the code when my fingers are tired from copy-pasting.